src/compute/lower.rs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Defines kernel to extract a lower case of a \[Large\]StringArray
+
+use crate::array::*;
+use crate::{
+    datatypes::DataType,
+    error::{ArrowError, Result},
+};
+
+fn utf8_lower<O: Offset>(array: &Utf8Array<O>) -> Utf8Array<O> {
```

Comment:
Given that there are many possible functions on a `&str`. 

Maybe we can make a `utf8_apply` that is generic over the function applied to the utf8 array.

This would not incur any compilation cost until implemented. 

We could the  simply document the `lower` and `higher`, or add them under a pre-compiled feature flag.

Reply:
I will give it a try

Reply:
that is a good idea; it makes it very easy for others to add new kernels.

Reply:
Addressed, I'm not sure whether this is what you want, PTAL.